homeassistant/components/serial/sensor.py

```diff
@@ -65,26 +64,42 @@ def __init__(self, name, port, baudrate, value_template):
 
     async def serial_read(self, device, rate, **kwargs):
         """Read the data from the port."""
-        reader, _ = await serial_asyncio.open_serial_connection(
-            url=device, baudrate=rate, **kwargs
-        )
+        loggedError = False
         while True:
-            line = await reader.readline()
-            line = line.decode("utf-8").strip()
-
             try:
-                data = json.loads(line)
-                if isinstance(data, dict):
-                    self._attributes = data
-            except ValueError:
-                pass
-
-            if self._template is not None:
-                line = self._template.async_render_with_possible_json_value(line)
-
-            _LOGGER.debug("Received: %s", line)
-            self._state = line
-            self.async_schedule_update_ha_state()
+                reader, _ = await serial_asyncio.open_serial_connection(
+                    url=device, baudrate=rate, **kwargs
+                )
+                _LOGGER.info("Serial device %s connected", device)
+                while True:
+                    try:
+                        line = await reader.readline()
+                        line = line.decode("utf-8").strip()
+
+                        try:
+                            data = json.loads(line)
+                            if isinstance(data, dict):
+                                self._attributes = data
+                        except ValueError:
+                            pass
+
+                        if self._template is not None:
+                            line = self._template.async_render_with_possible_json_value(line)
+
+                        _LOGGER.debug("Received: %s", line)
+                        self._state = line
+                        self.async_schedule_update_ha_state()
```

Comment:
This can be replaced by `self.async_write_ha_state()` because there is no update function.
```suggestion
                        self.async_write_ha_state()
```